contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -4455,26 +4454,34 @@ bbf_pivot(PG_FUNCTION_ARGS)
 		ereport(ERROR,
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("materialize mode required, but it is not allowed in this context")));
-
-	/* 
-	* Previously we saved two raw parsetrees in tsql outermost context
-	* here we are retrieve those raw parsetree for pivot execution
-	*/
-	tsql_outmost_estat = get_outermost_tsql_estate(&nestlevel);
-	tsql_outmost_context = tsql_outmost_estat->stmt_mcontext_parent;
 	
-	if (!tsql_outmost_context)
+	if (fcinfo->context == NULL || !IsA(fcinfo->context, List) || list_length((List *) fcinfo->context) != 3)
+		ereport(ERROR,
+			(errcode(ERRCODE_CHECK_VIOLATION),
+				errmsg("Babelfish PIVOT is not properly initialized.")));
+
+	node = list_nth((List *)fcinfo->context, 0);
+	if (!IsA(node, List) 
+			|| !IsA(list_nth((List *)node, 0), String) 
+			|| strcmp(((String *)list_nth((List *)node, 0))->sval, "bbf_pivot_func") != 0)
+	{
 		ereport(ERROR,
-			(errcode(ERRCODE_SYNTAX_ERROR),
-				errmsg("pivot outer context not found")));
+			(errcode(ERRCODE_CHECK_VIOLATION),
+				errmsg("Babelfish PIVOT is not properly initialized.")));
+	}
+	pivot_parsetree = (List *) list_nth((List *) fcinfo->context, 1);
```

Comment:
Since we're reusing fcinfo->context, pls add some test case to : 
1. trigger, inside trigger procedure run a pivot query
2. table function, in the def of the function, pls run a pivot query

Reply:
Thanks for the comments! 
1. Will add a trigger test. 
2. We already have a table function test cases which is define in `pivot-vu-prepare.sql:433`.

https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/45d60a99620d323660f4e3f77578de6ad3fb43d1/test/JDBC/input/pivot-vu-prepare.sql#L443